### PR TITLE
Status command

### LINKED ISSUE #32: Command:  status
This exposes node details , like nodeId and publicKey/eth address

TO DO:

- [ ]  implement core function status
- [ ]   expose it on http interface
- [ ]   expose it on p2p interface

Command Description:

```
{
command: "status",
node: "NodeB ID", // optional, if missing, it refers to current node
}
```

Response JSON: 
```
{
"id": "1234",
"publicKey": "0x123",
"address": "0x123",
"version": "v1.0",
"http" : true/false,
"p2p": true/false,
"provider": [
     {
        "chainId" : "5123",
        "network": "polygon",
     },
    { 
        "chainId" : "53",
        "network": "eth",
    }
],
"indexer": 
 [
    {
        "chainId" : "5123",
        "network": "polygon",
        "block":"123"
    },
    { 
        "chainId" : "53",
        "network": "eth",
        "block":"123"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes #32  .

Changes proposed in this PR:

- [x] implemented core function for `status`
- modified `OceanNodeKeys` to include `ethAddress`
- generated address from private key in `getPeerIdFromPrivateKey`

- [x] expose HTTP interface
- [x] expose P2P interface

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -30,3 +30,21 @@ export const SUPPORTED_PROTOCOL_COMMANDS: string[] = [
   PROTOCOL_COMMANDS.GET_DDO,
   PROTOCOL_COMMANDS.NONCE
 ]
+
+export const SUPPORTED_NETWORK_NAMES_BY_CHAIN_IDS: any = {
```
Comment (mariacarmina) [inline src/utils/constants.ts:34]: I'll import address.json file when it'll be marked ready for review.

**Comment by mariacarmina** [comment]
This is the response from Postman calling `/directCommand` with status:

Screenshot 2023-11-20 at 13 59 33

Extended version:

```json
{
    "id": "16Uiu2HAmJKKnHoDrc5xcxPfVJUWLrv8iNnhqZ3YdnUfRRsAiJvAJ",
    "publicKey": {
        "0": 8,
        "1": 2,
        "2": 18,
        "3": 33,
        "4": 3,
        "5": 84,
        "6": 28,
        "7": 121,
        "8": 227,
        "9": 133,
        "10": 2,
        "11": 175,
        "12": 121,
        "13": 26,
        "14": 211,
        "15": 103,
        "16": 80,
        "17": 5,
        "18": 222,
        "19": 172,
        "20": 81,
        "21": 16,
        "22": 222,
        "23": 176,
        "24": 207,
        "25": 4,
        "26": 178,
        "27": 204,
        "28": 198,
        "29": 186,
        "30": 188,
        "31": 60,
        "32": 16,
        "33": 205,
        "34": 204,
        "35": 13,
        "36": 231
    },
    "address": "0x02354A1F160A3fd7ac8b02ee91F04104440B28E7",
    "version": "1.0.0",
    "http": true,
    "p2p": true,
    "provider": [
        {
            "chainId": "1",
            "network": "mainnet"
        },
        {
            "chainId": "137",
            "network": "polygon"
        },
        {
            "chainId": "80001",
            "network": "mumbai"
        }
    ],
    "indexer": [
        {
            "chainId": "1",
            "network": "mainnet",
            "block": "0"
        },
        {
            "chainId": "137",
            "network": "polygon",
            "block": "0"
        },
        {
            "chainId": "80001",
            "network": "mumbai",
            "block": "0"
        }
    ]
}

```
For testing, I suppose [this issue](https://github.com/oceanprotocol/ocean-node/issues/117) needs to be solved first in order to create an Ocean node class and call this commands when the node is up.
cc @alexcos20

File: src/components/P2P/index.ts [hunk 2]
```diff
@@ -434,6 +435,15 @@ export class OceanP2P extends EventEmitter {
     return peersFound
   }
 
+  async getStatus(nodeId: string) {
+    console.log('Get status for ' + nodeId)
+    try {
+      return await status(nodeId)
+    } catch (error) {
+      console.error(error)
```
Comment (jamiehewitt15) [inline src/components/P2P/index.ts:443]: Why aren't you using the Logger here?

Comment (mariacarmina) [reply src/components/P2P/index.ts:443]: This function needs to be removed. I'll do that immediatly.

**Review by paulo-ocean — COMMENTED** [review]
hi @mariacarmina 
overall, lgtm , just wondering if we should do the public key more "user friendly" to read.. instead of returning the byte array we could do something like
`Buffer.from(config.keys.publicKey).toString('hex')` for instance
Like we have on the node `getPeerDetails()`
Just cosmetic anyway, but imo, it might look better & is more aligned with the other fields.
thanks

**Comment by mariacarmina** [comment]
> hi @mariacarmina 
> 
> overall, lgtm , just wondering if we should do the public key more "user friendly" to read.. instead of returning the byte array we could do something like
> 
> `Buffer.from(config.keys.publicKey).toString('hex')` for instance
> 
> Like we have on the node `getPeerDetails()`
> 
> Just cosmetic anyway, but imo, it might look better & is more aligned with the other fields.
> 
> thanks
> 
> 

Sure thing, I'll do that to look easier to read. Thanks!

**Comment by mariacarmina** [comment]
Updated response

Screenshot 2023-11-20 at 22 25 01

**Review by paulo-ocean — APPROVED** [review]

**Review by alexcos20 — APPROVED** [review]

File: src/utils/blockchain.ts [hunk 3]
```diff
@@ -28,4 +30,24 @@ export class Blockchain {
   public getSupportedChains(): string[] {
     return this.supportedChains
   }
+
+  public getNetworkNameByChainId(chainId: string): string {
+    let networkName: string
+    const addressFile = JSON.parse(
+      // eslint-disable-next-line security/detect-non-literal-fs-filename
+      fs.readFileSync(
```
Comment (alexcos20) [inline src/utils/blockchain.ts:38]: can you add a small comment like '// TODO - change me' here ?
We will change this once indexer is merged

Comment (mariacarmina) [reply src/utils/blockchain.ts:38]: Ok, sure, I'll add it. Thanks!

Comment (bogdanfazakas) [reply src/utils/blockchain.ts:38]: we are going to have this in config, so no need to add this method in the blockchain helper class

Comment (mariacarmina) [reply src/utils/blockchain.ts:39]: We are going to modify once indexer is merged, too.

**Pull request merged**